src/openrct2/ride/Track.cpp

```diff
@@ -244,6 +244,7 @@ bool track_add_station_element(CoordsXYZD loc, ride_id_t rideIndex, int32_t flag
                     targetTrackType = TrackElemType::MiddleStation;
                 }
                 stationElement->AsTrack()->SetTrackType(targetTrackType);
+                stationElement->AsTrack()->SetRideType(ride->type);
```

Comment:
I'm not sure this would be what the user expects. Imagine this scenario: 

You build a station and change its track type (ride type). Then you decide you want to expand the station but with a different track type (ride type). This 'helpfully' resets the track type of the original station end to the rides default.